### DISCUSSION TITLE
How do references work in this repo? Dependent project cannot find new interface added in parent

### DISCUSSION THREAD

**Discussion by IanKemp** [dotnet/aspnetcore#27206] [Q&A] [answered]
I cloned the repo @ 3.1.9, followed `BuildFromSource.md`, [got punched in the face a few times](https://github.com/dotnet/aspnetcore/issues/27171), but was finally able to make a  build of 3.1.9.

However, things still don't seem to work quite right. I added a new public interface in `Microsoft.AspNetCore.Mvc.Abstractions` and want to implement it in `Microsoft.AspNetCore.Mvc.Core`, which references the former - but when I type the interface's name in a class file in `Core`, it isn't found and Visual Studio prompts me to add a reference to `Abstractions` - which is already referenced???

The DLL for `Abstractions` that `Core` points to does contain my new interface (verified with ILSpy). `ReferenceResolution.md` doesn't note this as a pain point.

So I have no idea what's going on, and since I have zero experience with this repo (and getting it to even build the correct version was already an exercise in pain) I don't just want to add a project ref to `Abstractions`. Assistance requested.

**Comment by IanKemp** [comment]
Grabbed the repo code @ 3.1.10 but this issue persists:

* `IMyInterface` is defined in `Microsoft.AspNetCore.Mvc.Abstractions`
* `Microsoft.AspNetCore.Mvc.Abstractions` builds successfully
* `Microsoft.AspNetCore.Mvc.Core` references the built `Microsoft.AspNetCore.Mvc.Abstractions.dll` that I have verified to contain `IMyInterface`
* But, `Microsoft.AspNetCore.Mvc.Core` does not recognise `IMyInterface`

**Why not?**

**Comment by Pilchie** [comment]
@dougbu - any ideas here?

**Comment by dougbu** [comment, marked as answer]
The Microsoft.AspNetCore.Mvc.Core project references the ref/ assembly for Microsoft.AspNetCore.Mvc.Abstractions and that has not been updated according you your description. Run the following in the src/Mvc/Mvc.Abstractions/src/ folder:
``` powershell
dotnet msbuild /t:GenerateReferenceSource
```

See <https://github.com/dotnet/aspnetcore/blob/release/3.1/docs/ReferenceAssemblies.md> for more information.

**Reply by IanKemp** [reply]
Thanks @dougbu - after running `dotnet msbuild /t:GenerateReferenceSource` I found that I also had to run `.\build.cmd` in the repo root before Visual Studio would accept that my new interface did indeed exist.

**Reply by IanKemp** [reply]
I also had to manually edit some of the `*.Manual.cs` files in the `Mvc.*\ref\` subdirectories for other parts of the project to be happy.
